added_mass/proposal.md

```diff
@@ -0,0 +1,265 @@
+software. That also makes them easier to introspect and change independently of
+other inertial terms.
+
+### XML spec
+
+The [`//link/inertial`](http://sdformat.org/spec?ver=1.9&elem=link#link_inertial) term
+in SDF currenly accepts elements to describe the m matrix (mass and moments of inertia),
+as well as a center of mass pose. This proposal suggests exposing each of the 21 elements
+for the mass matrix in addition to that. All added mass elements are optional and default
+to zero if unset. This preserves the behaviour for links that don't have those terms.
+
+A new `<fluid_added_mass>` element will be added under `//link/inertial/`.
+It will contain each of the 21 matrix elements.
+
+```xml
+<inertial>
+  <mass>10</mass>
+  <pose>1 1 1 0 0 0</pose>
+  <inertia>
+    <ixx>0.16666</ixx>
+    <ixy>0</ixy>
+    <ixz>0</ixz>
+    <iyy>0.16666</iyy>
+    <iyz>0</iyz>
+    <izz>0.16666</izz>
+  </inertia>
+  <fluid_added_mass>
+    <xx>1.0</xx>
+    <xy>0.0</xy>
+    <xz>0.0</xz>
+    <xp>0.0</xp>
+    <xq>0.0</xq>
+    <xr>0.0</xr>
+    <yy>1.0</yy>
+    <yz>0.0</yz>
+    <yp>0.0</yp>
+    <yq>0.0</yq>
+    <yr>0.0</yr>
+    <zz>1.0</zz>
+    <zp>0.0</zp>
+    <zq>0.0</zq>
+    <zr>0.0</zr>
+    <pp>0.1</pp>
+    <pq>0.0</pq>
+    <pr>0.0</pr>
+    <qq>0.1</qq>
+    <qr>0.0</qr>
+    <rr>0.1</rr>
+  </fluid_added_mass>
+</inertial>
+```
+
+### C++ API
+
+The `//link/inertial` element maps to a
+[gz::math::Inertiald](https://gazebosim.org/api/math/6.10/classignition_1_1math_1_1Inertial.html)
+object. To support added mass, the following member functions will be added to the
+`Inertial` class:
+
+```cpp
+/// \brief Constructs an inertial object from the mass matrix for a body
+/// B, about its center of mass Bcm, and expressed in a frame that we’ll
+/// call the "inertial" frame Bi, i.e. the frame in which the components
+/// of the mass matrix are specified (see this class’s documentation for
+/// details). The pose object specifies the pose X_FBi of the inertial
+/// frame Bi in the frame F of this inertial object
+/// (see class’s documentation). The added mass matrix is also expressed
+/// in frame Bi.
+/// \param[in] _massMatrix Mass and inertia matrix.
+/// \param[in] _pose Pose of center of mass reference frame.
+public: Inertial(const MassMatrix3<T> &_massMatrix,
+                 const Pose3<T> &_pose,
+                 const Matrix6<T> &_addedMass)
+
+/// \brief Set the added mass matrix.
+///
+/// \param[in] _m New Matrix6 object, which must be a symmetric matrix.
+/// \return True if the Matrix6 is valid.
+public: bool SetFluidAddedMass(const Matrix6<T> &_m)
+
+/// \brief Get the added mass matrix.
+/// \return The added mass matrix.
+public: const Matrix6<T> &FluidAddedMass() const
+
+/// \brief Spatial mass matrix, which is the resulting 6x6 matrix including
+/// added mass.
+/// \return The spatial mass matrix.
+public: const Matrix6<T> &SpatialMatrix() const
```

Comment:
thanks for your patience; this draft of the proposal looks good so far. I think we just need to clarify the order of variables used in this spatial matrix. The matrix earlier in this document uses `[x y z p q r]` ordering, but I believe dartsim and drake have the rotational variables first `[p q r x y z]`. We don't need to arbitrary parameter ordering but just document the order

Reply:
I expect that variable ordering convention is used in dartsim and drake because it is the one used by Featherstone:

* https://homes.cs.washington.edu/~todorov/courses/amath533/FeatherstoneSlides.pdf

Reply:
Updated